### PR TITLE
[#1335] Skip invalid responses

### PR DESCRIPTION
**Pull request by ichinaski**
* Don't process SV w/o a valid questionID
* Simplify type assignment

### CODE REVIEW

**Comment by iperdomo** [comment]
## Checklist

* [ ] Test plan
* [ ] Copyright header
* [ ] Code formatting

File: GAE/src/org/waterforpeople/mapping/app/web/SurveyedLocaleServlet.java [hunk 1]
```diff
@@ -182,19 +179,16 @@ protected SurveyedLocaleResponse convertToResponse(List<SurveyedLocale> slList,
...
+                                questionTypeMap.put(sv.getSurveyQuestionId(), type);
                             }
                         }
                     }
-                    if (!sv.getQuestionType().equals("IMAGE")) {
-                        // add question type
-                        siDto.addProperty(sv.getSurveyQuestionId(),
-                                sv.getStringValue() != null ? sv.getStringValue() : "",
-                                deviceQuestionType);
-                    }
```
Comment (ichinaski) [inline GAE/src/org/waterforpeople/mapping/app/web/SurveyedLocaleServlet.java]: We should probably double check this change with @mtwestra, as I fail to see the reason for having this check in the first place.

Comment (muloem) [inline GAE/src/org/waterforpeople/mapping/app/web/SurveyedLocaleServlet.java]: We do sync images as well now right?

Comment (ichinaski) [inline GAE/src/org/waterforpeople/mapping/app/web/SurveyedLocaleServlet.java]: We always sync images, yes. The bit that kept me puzzled was that a `SurveyalValue`'s question type, as we saw the other day, is always *PHOTO*, not *IMAGE*, which is exactly the mapping happening a few lines above. So my concerns with that conditional check were:
* If question type is always *PHOTO*, the condition *always* holds true.
* Even if the question type is set to *IMAGE*, why would we cut it out from the responses?

Comment (muloem) [inline GAE/src/org/waterforpeople/mapping/app/web/SurveyedLocaleServlet.java]: Perhaps in previous versions of the app images were not synced?

Comment (ichinaski) [inline GAE/src/org/waterforpeople/mapping/app/web/SurveyedLocaleServlet.java]: Image *responses* have always been synced. Images themselves were synced in more recent versions, but this feature is completely unrelated.

**Comment by muloem** [comment]
:+1: code reviewed!
